### PR TITLE
split code for utility/testing; add js testing

### PR DESCRIPTION
**Pull request by AIasd**

### CODE REVIEW

File: lemur/tests/test_integration_db_and_models.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
+# This file consists of testing for the creation of objects and their
+# relationship in db
+import sys
+sys.path.append('../..')
```
Comment (AWinterman) [inline lemur/tests/test_integration_db_and_models.py]: Why is this needed? This is going to change your path each time this gets run, won't it?

Comment (AIasd) [inline lemur/tests/test_integration_db_and_models.py]: I want to separate testing scripts from the files to be tested so I have to change the path in order to access the files to be tested.

File: lemur/tests/tests_script.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# This script is used to run all the test files in the tests folder
+import os
+
+
+def run_tests():
+    os.system('python3 test_integration_utility_generate_and_convert.py')
```
Comment (AWinterman) [inline lemur/tests/tests_script.py]: woah, you might check out nose, which is a test runner for python.

File: lemur/tests/helper_random.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+import sys
+sys.path.append('../..')
+# Libraries
+# Standard library
+from random import choice, randint, shuffle
+from string import ascii_lowercase, ascii_uppercase, digits
+
+# Local
+from lemur import models as m
+from lemur.utility.utility_generate_and_convert import (generate_lab_id,
```
Comment (AWinterman) [inline lemur/tests/helper_random.py]: naming nit: there's no need to repeat utility here, from lemure.utility.generate_and_convert has the same information as lemur.utility.utility_generate_and_convert

File: lemur/tests/js_test.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    numberEntry('0.8','1.0-10.0',false);
+    textEntry('Critical','Critical,Helpful',true);
+    textEntry('Criticall','Critical,Helpful',false);
+});
+
+QUnit.module( 'group student_select_lab' );
+QUnit.test('decomposeClassId', function(assert) {
+    var classId = 'BIOL101_FALL2016';
+    assert.deepEqual(decomposeClassId(classId), ['BIOL101','FALL2016']);
+    assert.notDeepEqual(decomposeClassId(classId), undefined);
```
Comment (AWinterman) [inline lemur/tests/js_test.js]: these statements are redundant with your previous one. You are guaranteed that it is not undefined because you know it is `['BIOL101','FALL2016']`

File: lemur/tests/js_test.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    assert.notDeepEqual(decomposeClassId(classId), ['BIOL101','SPRING2016']);
+});
+QUnit.test('generateLabId', function(assert) {
+    var labName = 'Cortisol2_Carey_Thursday';
+    var classId = 'BIOL101_FALL2016';
+
+    assert.deepEqual(generateLabId(labName, classId), labName+':'+classId);
+    assert.notDeepEqual(generateLabId(labName, classId), labName+'_'+classId);
```
Comment (AWinterman) [inline lemur/tests/js_test.js]: same nit about redundant asserts.

File: lemur/tests/js_test.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    assert.notDeepEqual(generateLabId(labName, classId), undefined);
+});
+
+QUnit.module( 'group admin_edit_data' );
+QUnit.test('decomposeClassId', function(assert) {
+    var experimentId = 'Cortisol2_Carey_Thursday:BIOL101_FALL2016:Cortisol Concentration';
+    var studentName = 'amyth';
+
+    assert.deepEqual(generateObservationId(experimentId, studentName), experimentId+':'+studentName);
+    assert.notDeepEqual(generateObservationId(experimentId, studentName), experimentId+'_'+studentName);
```
Comment (AWinterman) [inline lemur/tests/js_test.js]: and here too.

**Comment by AWinterman** [comment]
a couple of nits about redundant asserts, and you should really use a test runner like nose, which i think will let you avoid any path manipulations and let you stop making system calls to run your tests, but overall, lgtm.

File: lemur/tests/helper_random.py (hunk 3 above)
Comment (AIasd) [inline lemur/tests/helper_random.py]: I will fix them once I settled down the path of the files switching to another testing framework.

File: lemur/tests/test_integration_db_and_models.py (hunk 1 above)
Comment (AWinterman) [inline lemur/tests/test_integration_db_and_models.py]: I really don't think that should be necessary, and part of what makes me think you should be investigating a test runner like nose.
